Crashlytics/Crashlytics/Models/FIRCLSInternalReport.m

```diff
@@ -75,9 +80,55 @@ - (instancetype)initWithPath:(NSString *)path executionIdentifier:(NSString *)id
 
   _identifier = [identifier copy];
 
+  [self _checkAndWriteOOMOfRequired];
+
   return self;
 }
 
+// Load the reports internal kv store
+- (NSDictionary<NSString *, NSString *> *)_loadInternalBreadcrumbs {
+  NSString *path = [self.path stringByAppendingPathComponent:FIRCLSReportInternalIncrementalKVFile];
+  NSArray *sections = FIRCLSFileReadSections(path.UTF8String, true, ^NSObject *(id obj) {
+    NSDictionary *dict = [obj objectForKey:@"kv"];
+    NSString *key = FIRCLSFileHexDecodeString(((NSString *)dict[@"key"]).UTF8String);
+    NSString *value = FIRCLSFileHexDecodeString(((NSString *)dict[@"value"]).UTF8String);
+    return (key && value) ? @{key : value} : @{};
+  });
+  NSMutableDictionary *res = [NSMutableDictionary dictionary];
+  for (NSDictionary *kv in sections) {
+    [res addEntriesFromDictionary:kv];
+  }
+  return [res copy];
+}
+
+// An OOM is pretty simple.
+// If the data in this report shows an OOM,
+// then write an exception file to disk.
+// That will be picked up by the normal reporting system.
+//
+// NOTE:
+// I'd like to write a signal SIGKILL file, or some other
+// kind of OOM file, but up to now I have not been able to have
+// it show up in the Firebase crashlytics console. As a matter of fact,
+// this one doesn't show up either. There must be something blocking
+// it server side.
+// see: https://github.com/firebase/firebase-ios-sdk/discussions/12897
+- (void)_checkAndWriteOOMOfRequired {
+  NSString *path = [self pathForContentFile:FIRCLSReportExceptionFile];
+  if ([NSFileManager.defaultManager fileExistsAtPath:path]) {
+    return;
+  }
+
+  // first check if we need to build one
+  // we look for all internal breabcrumbs
+  NSDictionary<NSString *, NSString *> *breadcrumbs = [self _loadInternalBreadcrumbs];
+  FIRCLSAppMemory *memoryInfo = [[FIRCLSAppMemory alloc] initWithJSONObject:breadcrumbs];
+  if (memoryInfo.isOutOfMemory) {
+    FIRCLSInfoLog(@"Writing OOM record to %@", path);
+    FIRCLSExceptionRecordOutOfMemoryTerminationAtPath(path.UTF8String);
```

Comment:
I'm surprised that even though the clsrecord is written out correctly, recognized by the local system as a valid part of the report and uploaded to the backend, it is not recognized server side and doesn't end up in the Firecase Crashlytics dashboard. Maybe the maintainers could help me figure out why (I don't think I have access to the backend code).

Reply:
@paulb777 @ncooke3 do either of you have any ideas how why this type of report won't go through on the backend (or who I might refer to)?